scripts/generate_example_data.py

```diff
@@ -58,25 +60,82 @@ def main(output_dir):
 def generate_patient_data():
     # Generate some random data for our patient
     sex = random.choice(["male", "female"])
+    probability = ["true", "false"]
+    is_dead = random.choice(probability)
+    probability.remove(is_dead)
     date_of_birth = random_date(date(1950, 1, 1), TODAY)
     # Round date of birth to the first of the month which reflects what happens in the
     # real data
     date_of_birth = date_of_birth.replace(day=1)
 
+    if is_dead == "true":
+        date_of_death = random_date(date_of_birth, TODAY)
+    else:
+        date_of_death = None
+
     # You can think of `yield` a bit like `return` except that we can call it multiple
     # times in the same function. This makes it easier for our function to provide data
     # for multiple different tables.
-    yield "patients", {"sex": sex, "date_of_birth": date_of_birth}
+    yield "patients", {"sex": sex, "date_of_birth": date_of_birth, "date_of_death": date_of_death}
 
     # Decide how many medications should our patient be issued
     medications_count = random.randrange(0, 10)
 
     # Generate data for each of these medication issues
     for _ in range(medications_count):
-        meds_date = random_date(date_of_birth, TODAY)
+        if date_of_death == None:
+            meds_date = random_date(date_of_birth, TODAY)
+        else:
+            meds_date = random_date(date_of_birth, date_of_death)
         meds_dmd_code = random.choice(DMD_CODES)
         yield "medications", {"date": meds_date, "dmd_code": meds_dmd_code}
 
+    # Check when the patient registered in the practice
+    # set the max number of registrations per patient
+    registrations_count = random.randrange(1, 11)
+
+    # generate data for registrations
+    generated_dates = []
+    for _ in range(registrations_count):
+        practice_pseudo_id = random.choice(pseudo_id)
+        if date_of_death == None:
```

Comment:
Instead of two big blocks that are very similar, I'd do:

```
if is_dead:
    last_possible_date = date_of_death
else:
    last_possible_date = TODAY
```

and then just have one block, and do:

```
    start_date = random_date(date_of_birth, last_possible_date)
    end_date = random_date(start_date, last_possible_date)
```